integrations/connext/components/form-connext-xtransfer.tsx

```diff
@@ -0,0 +1,505 @@
+import { useLatestTransfers } from '../hooks/use-latest-transfers'
+import { useSupportedTransfer } from '../hooks/use-supported-transfer'
+import { useXcall } from '../hooks/use-xcall'
+import { mainnetAssets, testnetAssets } from '../utils/assets'
+import { mainnetChains, testnetChains } from '../utils/chains'
+import { arbitrumDomainId, arbitrumGoerliId, optimismDomainId, optimismGoerliDomainId } from '../utils/constants'
+import { Asset } from '../utils/types'
+
+interface FormConnextXTransferProps {
+  isMainnet: boolean
+  setIsMainnet: Dispatch<SetStateAction<boolean>>
+}
+
+export function FormConnextXTransfer({ isMainnet, setIsMainnet }: FormConnextXTransferProps) {
+  const { chain } = useNetwork()
+  const { address } = useAccount()
+  const { isLoading: isSwitchingChain, switchNetwork } = useSwitchNetwork()
+
+  const [originChain, setOriginChain] = useState(optimismDomainId)
+  const [destinationChain, setDestinationChain] = useState(arbitrumDomainId)
+  const [asset, setAsset] = useState('eth')
+  const [amount, setAmount] = useState<string>('')
+  const [showTransferStatus, setShowTransferStatus] = useState(false)
+  const [contractApproved, setContractApproved] = useState(false)
+
+  const getChain = (domainId: string) => {
+    return isMainnet
+      ? mainnetChains.find((mainnetChain) => mainnetChain.domain_id == domainId)
+      : testnetChains.find((testnetChain) => testnetChain.domain_id == domainId)
+  }
+  const getAsset = (): Asset | undefined => {
+    return isMainnet
+      ? mainnetAssets.find((mainnetAsset) => mainnetAsset.id === asset)
+      : testnetAssets.find((testnetAsset) => testnetAsset.id === asset)
+  }
+  const { data: originBalance } = useBalance({
+    address,
+    token: getAsset()?.contracts.find((contract) => contract.chain_id === getChain(originChain)?.chain_id)?.contract_address as `0x${string}`,
+    chainId: getChain(originChain)?.chain_id ?? 1,
+  })
+  const { data: destinationBalance } = useBalance({
+    address,
+    token: getAsset()?.contracts.find((contract) => contract.chain_id === getChain(destinationChain)?.chain_id)?.contract_address as `0x${string}`,
+    chainId: getChain(destinationChain)?.chain_id ?? 1,
+  })
+
+  const transferSupported = useSupportedTransfer({
+    originChainId: getChain(originChain)?.chain_id,
+    destinationChainId: getChain(destinationChain)?.chain_id,
+    assetDataContracts: getAsset()?.contracts,
+  })
+
+  const {
+    estimatedAmount,
+    isFastPath,
+    isLoading: isFetchingEstimatedAmount,
+  } = useEstimatedAmount({
+    isMainnet,
+    originDomain: getChain(originChain)?.domain_id,
+    destinationDomain: getChain(destinationChain)?.domain_id,
+    originTokenAddress: getAsset()?.contracts.find((contract) => contract.chain_id === getChain(originChain)?.chain_id)?.contract_address,
+    amount: BigInt(
+      Number(amount) * 10 ** (getAsset()?.contracts.find((contract) => contract.chain_id === getChain(originChain)?.chain_id)?.decimals ?? 18)
```

Comment:
use [`parseUnits`](https://viem.sh/docs/utilities/parseUnits.html) from Viem in these cases